R/meltAssay.R

```diff
@@ -0,0 +1,152 @@
+#' @title meltAssay
+#'
+#' @description Converts a \code{\link[=TreeSummarizedExperiment-class]{TreeSummarizedExperiment}} object into a
+#' long data.frame which can be used for \code{\link[tidyverse]{tidyverse}}-tools.
+#'
+#' @param x a numeric matrix or a \code{\link[=TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#'          object containing a tree.
+#' @param abund_values Must be one of the values returned by assayNames.
+#' @param add_col_data Choice of colData columns to add. Default="none".
+#' @param ... optional arguments not used.
+#' @return A data.frame tbl_df.
+#' @examples
+#' \dontrun{
+#' data(GlobalPatterns)
+#' data(GlobalPatterns, package = "MicrobiomeExperiment")
+#' se <- GlobalPatterns
+#' se.rel <- relAbundanceCounts(se)
+#' molten_se <- meltAssay(se.rel,
+#'   add_col_data = colnames(colData(se.rel)),
+#'   abund_values = "counts"
+#' )
+#' head(molten_se)
+#' }
+#' @importFrom SummarizedExperiment assayNames rowData rowData<-
+#' @importFrom tibble rownames_to_column
+#' @importFrom tidyr pivot_longer
+#' @importFrom dplyr left_join mutate_if %>%
+#' @keywords Utilities
+#' @export
+setGeneric("meltAssay",
+  signature = "x",
+  function(x, add_col_data = "none", abund_values = "counts", ...) {
+    standardGeneric("meltAssay")
+  }
+)
+
+
+#' @rdname meltAssay
+#' @aliases meltAssay
+#'
+#' @importFrom SummarizedExperiment rowData rowData<- colData colData<-
+#'
+#' @export
+setMethod("meltAssay",
+  signature = c(x = "TreeSummarizedExperiment"),
```

Comment:
It does not need to be a `TreeSummarizedExperiment`. Just a `SummarizedExperiment` would work as well, since `SCE` or `TSE` specific features are not used.